### PR TITLE
Update component to be compatible with VideoJS 8.0.4+

### LINKED ISSUE #604: `videojs.extend` has been deprecated
But it's been used [here](https://github.com/videojs/videojs-youtube/blob/v2.6.1/src/Youtube.js#L40).

### PR DESCRIPTION
**Pull request by mattporritt**
The videojs.extend() function is removed from Video.js 8.0 See: https://videojs.com/guides/videojs-7-to-8/#videojsextend This patch reworks code to be an ES6 Class.

Fixes #604

Plugin unit tests also pass, and have tested with a videojs 8.0.4 implementation

### CODE REVIEW

**Comment by astagi** [comment]
@gkatsev @eXon @revolunet I tried this PR and it works like a charm for VideoJS 8.0.4! Could you please merge and release a major version please? We're already using it in the Italian Design System https://github.com/italia/bootstrap-italia/pull/841

Thank you @mattporritt

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "videojs-youtube",
   "description": "YouTube playback technology for Video.js",
-  "version": "2.6.1",
```
Comment (gkatsev) [inline package.json:4]: can you revert this change?

File: dist/Youtube.js [hunk 2]
```diff
@@ -37,10 +37,10 @@ THE SOFTWARE. */
   var _isOnMobile = videojs.browser.IS_IOS || videojs.browser.IS_NATIVE_ANDROID;
```
Comment (gkatsev) [inline dist/Youtube.js:37]: can you revert the changes in `dist/`? They'll get updated as part of a release.

File: package.json [hunk 3]
```diff
@@ -34,14 +34,14 @@
...
+    "karma": "^1.7.1",
     "karma-chrome-launcher": "^2.0.0",
     "karma-jasmine": "^1.0.2",
     "mkdirp": "^0.5.1",
     "precommit-hook": "^3.0.0",
     "uglify-js": "^3.4.6"
   },
   "dependencies": {
-    "video.js": "^5.6.0 || ^6.2.8 || ^7.0.2"
+    "video.js": "5.x || 6.x || 7.x"
```
Comment (gkatsev) [inline package.json:45]: This needs `|| 8.x` to properly support v8.
```suggestion
    "video.js": "5.x || 6.x || 7.x || 8.x"
```

**Review by gkatsev — COMMENTED** [review]
Thank you!

I verified that this continues to work with older versions of Video.js as well as now working with Video.js 8.

Just had a couple of small asks!

I'd likely release this as a major because the new class syntax may break folks.

**Comment by mattporritt** [comment]
Hi @gkatsev ,
Thanks for your patience on this. I've made the changes you've requested. I've rewritten and forced pushed so everything is still in one commit.

Cheers,
Matt P

**Review by gkatsev — APPROVED** [review]
Thanks!

**Comment by gkatsev** [comment]
Published as 3.0.0!
